hail/python/hailtop/hailctl/auth/login.py

```diff
@@ -76,21 +76,18 @@ def init_parser(parser):  # pylint: disable=unused-argument
     token = resp['token']
     username = resp['username']
 
-    auth_ns = deploy_config.service_ns('auth')
-    tokens = get_tokens()
-    tokens[auth_ns] = token
     dot_hail_dir = os.path.expanduser('~/.hail')
     if not os.path.exists(dot_hail_dir):
         os.mkdir(dot_hail_dir, mode=0o700)
-    tokens.write()
+    auth_ns = deploy_config.service_ns('auth')
+    Tokens({auth_ns: token}).write()
```

Comment:
This is not OK.  Tokens potentially has tokens for multiple instances of auth, and you're discarding all the other ones.  You need to read/modify the existing tokes file.